aioshelly/__init__.py

```diff
@@ -144,9 +144,10 @@ def __init__(
         self._settings = None
         self.shelly = None
         self._status = None
-        self._unsub_listening = coap_context.subscribe_updates(
-            options.ip_address, self._coap_message_received
-        )
+        if coap_context:
```

Comment:
This is wrong. We always need a coap context or else you cannot fetch cit/s and cit/d.

Reply:
verify.py script in this repo doesn't use a coap context because it doesn't fetch live data but just validate vs json file of cit/d and cit/s.

Reply:
Then the verify script needs to pass in a mock object. We should not allow setting up our class without a coap object as it will break.

Reply:
```python
from unittest.mock import Mock

COAP = MocK()
```

Reply:
but that way a call to `coap_context.subscribe_updates()` still fails

Reply:
Push up the code that you have.